bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -1724,63 +1797,85 @@ private boolean verifyCookie(Cookie oldCookie, File dir) throws IOException {
-        }
+            if (!allVersionFiles.isEmpty()) {
+                if (delete) {
+                    boolean confirm = force;
+                    if (!confirm) {
+                        confirm = IOUtils.confirmPrompt("Are you sure you want to delete Cookies locally?");
+                    }
+                    if (confirm) {
+                        for (File verFile : allVersionFiles) {
+                            if (!verFile.delete()) {
+                                LOG.error(
+                                        "Failed to delete Local cookie file {}. So aborting deletecookie of Bookie: {}",
+                                        verFile, bookieAddress);
+                                return -1;
+                            }
+                        }
+                        LOG.info("Deleted Local Cookies of Bookie: {}", bookieAddress);
+                    } else {
+                        LOG.info("Skipping deleting local Cookies of Bookie: {}", bookieAddress);
+                    }
+                } else {
+                    LOG.info("Listing local Cookie Files of Bookie: {}", bookieAddress);
+                    for (File verFile : allVersionFiles) {
+                        LOG.info(verFile.getCanonicalPath());
+                    }
+                }
+            } else {
+                LOG.info("No local cookies for Bookie: {}", bookieAddress);
+            }
 
-        @Override
-        int runCmd(CommandLine cmdLine) {
-            ServerConfiguration conf = new ServerConfiguration(bkConf);
             try (RegistrationManager rm = new ZKRegistrationManager()) {
+                Versioned<Cookie> cookie = null;
                 try {
-                    rm.initialize(bkConf, () -> {
-                    }, NullStatsLogger.INSTANCE);
-                } catch (BookieException e) {
-                    LOG.error("Exception while establishing zookeeper connection.", e);
-                    return -1;
+                    rm.initialize(bkConf, () -> { }, NullStatsLogger.INSTANCE);
+                    cookie = Cookie.readFromRegistrationManager(rm, bookieAddress);
+                } catch (CookieNotFoundException nne) {
+                    LOG.info("No cookie for {} in ZooKeeper", bookieAddress);
+                    return 0;
                 }
 
-                List<File> allLedgerDirs = Lists.newArrayList();
-                allLedgerDirs.addAll(Arrays.asList(ledgerDirectories));
-                if (indexDirectories != ledgerDirectories) {
-                    allLedgerDirs.addAll(Arrays.asList(indexDirectories));
-                }
+                if (delete) {
+                    boolean confirm = force;
+                    if (!confirm) {
+                        confirm = IOUtils.confirmPrompt("Are you sure you want to delete Cookies from Zookeeper?");
+                    }
 
-                try {
-                    conf.setAllowStorageExpansion(true);
```

Comment:
I removed this line, not sure why this is added in this commit -https://github.com/sijie/bookkeeper/commit/0db0cf6181f4949fe4684770bb81442694a24b27

Reply:
@reddycharan I added this because this command is 'expandStorage'. so I set this flag explicitly, don't rely on the configuration file.

Reply:
hmm..I'm not sure if overriding conf value is the expected behavior.

Reply:
Are you guys the original contributor of `expandstorage`? If the intend is to use whatever value in the configuration, I am fine with removing overrides here. but it would be worth calling out in the help description of this command, because it relies on whatever value provided in the configuration.

Reply:
yes we are the original contributors.

Anyhow I reverted the removal..i added it back.